lib/ansible/vars/plugins.py

```diff
@@ -54,8 +56,22 @@ def get_vars_from_path(loader, path, entities, stage):
                 vars_plugin_list.append(vars_plugin)
 
     for plugin in vars_plugin_list:
-        if plugin._load_name not in C.VARIABLE_PLUGINS_ENABLED and getattr(plugin, 'REQUIRES_WHITELIST', False):
-            # 2.x plugins shipped with ansible should require enabling, older or non shipped should load automatically
+        # A collection plugin was enabled to get to this point because vars_loader.all() does not include collection plugins.
+
+        # legacy plugins always run by default, but they can set REQUIRES_ENABLED=True to opt out.
+        # This option does nothing for plugins in collections, but if it's set it should be True to reflect actual behavior.
+        legacy = '.' not in plugin._load_name
+        needs_enabled = not legacy
+        if hasattr(plugin, 'REQUIRES_WHITELIST'):
+            display.deprecated("The VarsModule class variable 'REQUIRES_WHITELIST' is deprecated. "
+                               "Use 'REQUIRES_ENABLED' instead.", version=2.18)
+            needs_enabled = plugin.REQUIRES_WHITELIST
+        elif hasattr(plugin, 'REQUIRES_ENABLED'):
```

Comment:
make it an if so 'REQUIRES_ENHABLED' has precedence

Reply:
Oh, good idea.